### ISSUE TITLE
Compilation fails in non-English locale settings

### ISSUE THREAD

**Issue by nukemiko** [Nuitka/Nuitka#2949]
- Nuitka version, full Python version, flavor, OS, etc. as output by *this exact* command.

  > python -m nuitka --version

    ```
  $ > python -m nuitka --version
    2.3.9
    Commercial: None
    Python: 3.12.4 (main, Jun  7 2024, 06:33:07) [GCC 14.1.1 20240522]
    Flavor: Arch Python
    Executable: /home/miketsu/Projects/PycharmProjects/Venvs/venv-3.12/bin/python
    OS: Linux
    Arch: x86_64
    Distribution: Arch None
    Version C compiler: /usr/bin/gcc (gcc 14.1.1).
    ```

- How did you install Nuitka and Python

  I installed Nuitka through pip in virtualenv.

- The specific PyPI names and versions

  It should be taken from this output if there specific packages involved, e.g. `numpy`, you are
  expected to shorten this to the relevant ones.

  > python -m pip list -v

  ```
  Package     Version Location                                                                            Installer
    ----------- ------- ----------------------------------------------------------------------------------- ---------
    Nuitka      2.3.9   /home/miketsu/Projects/PycharmProjects/Venvs/venv-3.12/lib/python3.12/site-packages pip
    ordered-set 4.1.0   /home/miketsu/Projects/PycharmProjects/Venvs/venv-3.12/lib/python3.12/site-packages pip
    pip         24.0    /home/miketsu/Projects/PycharmProjects/Venvs/venv-3.12/lib/python3.12/site-packages pip
    zstandard   0.22.0  /home/miketsu/Projects/PycharmProjects/Venvs/venv-3.12/lib/python3.12/site-packages pip
  ```

- Many times when you get an error from Nuitka, your setup may be special

  Try to compile anything, even a script to `print('Hello world')`.

- Also supply a Short, Self Contained, Correct, Example
  1. Create a python file named `script.py`, with the following content:
  ```python
  #!/bin/env python3
  print('Hello world')
  ```
  2. Compile `script.py` without any option, except the output filename:
  ```shell
  python -m nuitka script.py --output-filename=script.1.bin
  ```
  3. If the compilation was successful, check that the locale settings are `C.UTF-8` or `en_*.UTF-8` through command `locale`.
  4. Compile `script.py` again, use command line arguments same as step 2, but with a non-English `LC_MESSAGES` setting, such as `zh_CN.UTF-8`:
  ```shell
  LC_MESSAGES=zh_CN.UTF-8 LANGUAGE= python -m nuitka script.py --output-filename=script.2.bin
  ```
  5. Compilation will fail. You may see an error message like the following:
    ```
    Nuitka-Options: Used command line options: script.py --output-filename=script.1.bin
    Nuitka-Options:WARNING: You did not specify to follow or include anything but main program. Check options and make sure that is intended.
    Nuitka: Starting Python compilation with Nuitka '2.3.9' on Python '3.12' commercial grade 'not installed'.
    Nuitka: Completed Python level compilation and optimization.
    Nuitka: Generating source code for C backend compiler.
    Nuitka: Running data composer tool for optimal constant value handling.
    Nuitka: Running C compilation via Scons.
    Nuitka-Scons: Backend C compiler: gcc (gcc 14.1.1).
    Nuitka-Scons: Backend linking program with 6 files (no progress information available for this stage).
    /usr/bin/ld: 无效的 BFD 目标 None
    collect2: 错误：ld 返回 1
    scons: *** [/home/miketsu/Projects/PycharmProjects/Venvs/script.1.bin] Error 1
    FATAL: Failed unexpectedly in Scons C backend compilation.
    Nuitka:WARNING:     Complex topic! More information can be found at https://nuitka.net/info/scons-backend-failure.html
    Nuitka-Reports: Compilation crash report written to file 'nuitka-crash-report.xml'.
    ```

- Provide in your issue the Nuitka options used

  Only `--output-filename` in the command line.

- Addendum

  Compile with option `--standalone` or `--onefile` will not trigger this case.

**Comment by kayhayen** [comment]
I do not reproduce it, probably because I don't have that locale.

```
def initScons():
    # Avoid localized outputs.
    os.environ["LANG"] = "C"
    os.environ["LC_ALL"] = "C"
```

The second line is probably the fix for your setup, forcing Scons launched processes to ignore your language choices. As we sometimes parse its outputs, that much be more robust.

**Comment by kayhayen** [comment]
Thanks for your report, this is fixed on the factory branch, which is a development version under rapid development. You can try it out by going here: https://nuitka.net/doc/factory.html 

Feedback on whether this is working is very welcome. Please do not share plans to do it; only confirm or deny that it is working.

**Comment by nukemiko** [comment]
> Thanks for your report, this is fixed on the factory branch, which is a development version under rapid development. You can try it out by going here: https://nuitka.net/doc/factory.html
> 
> Feedback on whether this is working is very welcome. Please do not share plans to do it; only confirm or deny that it is working.

Confirmed that this issue has been fixed in the latest development version (2.4rc6).

**Comment by kayhayen** [comment]
This is part of the 2.3.10 hotfix I made.
